File: src/main/java/org/sonar/plugins/xquery/rules/XQueryRulesRepository.java
/*
 * © 2013 by Intellectual Reserve, Inc. All rights reserved.
 */

package org.sonar.plugins.xquery.rules;

import org.apache.commons.lang.StringUtils;
import org.sonar.api.profiles.RulesProfile;
import org.sonar.api.rules.*;
import org.sonar.api.utils.AnnotationUtils;
import org.sonar.api.utils.SonarException;
import org.sonar.check.Cardinality;
import org.sonar.plugins.xquery.api.XQueryConstants;
import org.sonar.plugins.xquery.checks.AbstractCheck;
import org.sonar.plugins.xquery.language.XQuery;

import java.lang.reflect.InvocationTargetException;
import java.util.ArrayList;
import java.util.List;
import java.util.logging.Logger;

/**
 * Repository for XQuery rules.
 * 
 * @since 1.0
 */
public final class XQueryRulesRepository extends RuleRepository {

    private static final String REPOSITORY_NAME = "XQuery";

    private final AnnotationRuleParser annotationRuleParser;

    public XQueryRulesRepository(AnnotationRuleParser annotationRuleParser) {
        super(CheckClasses.REPOSITORY_KEY, XQueryConstants.XQUERY_LANGUAGE_KEY);
        setName(REPOSITORY_NAME);
        this.annotationRuleParser = annotationRuleParser;
    }

    @Override
    public List<Rule> createRules() {
        return annotationRuleParser.parse(CheckClasses.REPOSITORY_KEY, CheckClasses.getChecks());
    }
}

File: src/main/java/org/sonar/plugins/xquery/rules/XQueryRulesRepository.kt
/*
 * © 2014 by Intellectual Reserve, Inc. All rights reserved.
 */

package org.sonar.plugins.xquery.rules

import org.sonar.api.rules.AnnotationRuleParser
import org.sonar.api.rules.Rule
import org.sonar.api.rules.RuleRepository
import org.sonar.plugins.xquery.api.XQueryConstants

/**
 * Repository for XQuery rules.
 *
 * @since 1.0
 */
class XQueryRulesRepository(private val annotationRuleParser: AnnotationRuleParser) : RuleRepository(CheckClasses.REPOSITORY_KEY, XQueryConstants.XQUERY_LANGUAGE_KEY) {

    init {
        name = REPOSITORY_NAME
    }

    override fun createRules(): List<Rule> {

        return annotationRuleParser.parse(CheckClasses.REPOSITORY_KEY, CheckClasses.checks)
    }

    companion object {
        private val REPOSITORY_NAME = "XQuery"
    }
}